float32[5] thetas
